renderlib/ScaleBarTool.cpp

```diff
@@ -0,0 +1,125 @@
+#include "ScaleBarTool.h"
+
+#include "ImageXYZC.h"
+#include "Logging.h"
+
+#include <iomanip>
+#include <sstream>
+
+// physicalscale is max of physical dims x,y,z
+static float
+computePhysicalScaleBarSize(const float physicalScale)
+{
+  // note this result will always be some integer power of 10 independent of zoom...
+  return pow(10.0f, floor(log10(physicalScale / 2.0f)));
+}
+
+static float
+getScaleBarWidthOrtho(int windowWidthPx, float windowPhysicalWidth)
+{
+  // We want to find the largest round number of physical units that keeps the scale bar within this width on screen
+  const int SCALE_BAR_MAX_WIDTH = (int)((float)windowWidthPx * 0.25f);
```

Comment:
Should this be in all-caps notation if it's just a local var?

Also nit, might be good to include units at the end of this variable name (`scaleBarMaxWidthPx`)

Reply:
Thanks, yeah. It used to be a true constant but not anymore. Will fix.